packages/sanes-chrome-extension/src/routes/welcome/index.tsx

```diff
@@ -1,6 +1,33 @@
 import * as React from 'react';
 import Layout from './components';
+import { ACCOUNT_STATUS_ROUTE } from '../paths';
+import { history } from '../../store/reducers/';
+import { PersonaContext } from '../../context/PersonaProvider';
+import { sendGetPersonaMessage } from '../../extension/messages';
 
-const Welcome = (): JSX.Element => <Layout />;
+const goToAccountStatus = (): void => {
+  history.push(ACCOUNT_STATUS_ROUTE);
+};
+
+const Welcome = (): JSX.Element => {
```

Comment:
You are using an external dependency `personaProvider` in a function (`onLoad`) which is declared outside of the effect, and this could lead into bugs. I suggest:
```
const personaProvider = React.useContext(PersonaContext);

React.useEffect(() => {
  const fetchPersona = async () => {
      const extensionContext = chrome !== ' undefined'
      if (!extensionContext) {
          return
      }
     const response = await sendGetPersonaMessage();
      if (response) {
        personaProvider.update(response.accounts, response.mnemonic, response.txs);
        goToAccountStatus();
      }
  }

  fetchPersona();
}, []);
````

Reply:
Your version does exactly the same thing as mine, just with the function declared in the `React.useEffect(` body.

The `onLoad` is captured in the `React.useEffect`, so no bug happens there.

Reply:
This is just part of the react hooks guidelines for avoiding potential bugs due to possible future changes. I believe I saw some examples where the personaProvider could not be the latest one (in case it has changed) I think lint should complain. 

I also think there is no problem because you use `[]` which means, do that on load. But I also think this does not harm anyone and we make react happy. What do you think?

Reply:
> I also think there is no problem because you use [] which means, do that on load. But I also think this does not harm anyone and we make react happy. What do you think?

The dependency `[]` means: no dependencies, run one on load. This is exactly what I want to do (and why the method is called `onLoad`: it should be run exactly once when the component is loaded.

If you want, I can move `async function onLoad() { ...}` into the `React.useEffect(` body like you did with `fetchPersona`. But that should not change anything.

Reply:
Just for following the react guidelines, please do.